internal/stage7.go

```diff
@@ -12,29 +13,34 @@ import (
 	"github.com/codecrafters-io/tester-utils/test_case_harness"
 )
 
-func getPath(executable string) string {
+func getPath(executable string, exePath string) string {
+	if executable == "my_exe" {
+		return exePath
+	}
+
 	path, err := exec.LookPath(executable)
 	if err != nil {
 		return fmt.Sprintf(`%s[:]? not found`, executable)
-	} else {
-		return path
 	}
+
+	return path
 }
 
 func testType2(stageHarness *test_case_harness.TestCaseHarness) error {
-	// Add the current directory to PATH (That is where the my_exe file is created)
-	//
-	// TODO: Remove this since it mutates path for ALL stages! Use shell.Setenv() instead.
-	//       We'll need to change the test to not use exec.LookPath for my_exe at least.
-	//       Also, always create my_exe in a RANDOM directory, not the current directory.
-	homeDir, _ := os.Getwd()
-	path := os.Getenv("PATH")
-	os.Setenv("PATH", fmt.Sprintf("%s:%s", homeDir, path))
+	// Add the random directory to PATH (where the my_exe file is created)
 
+	randomDir, err := GetRandomDirectory()
+	if err != nil {
+		return err
+	}
+
+	path := os.Getenv("PATH")
 	logger := stageHarness.Logger
 	shell := shell_executable.NewShellExecutable(stageHarness)
+	shell.Setenv("PATH", fmt.Sprintf("%s:%s", randomDir, path))
 
-	err := custom_executable.CreateExecutable(GetRandomString(), "my_exe")
+	exePath := filepath.Join(randomDir, "my_exe")
```

Comment:
```suggestion
	customExecutablePath := filepath.Join(randomDir, "my_exe")
```

(the way this is "different" from other executables we're testing is that it is a "custom" executable. Also, avoid abbreviations in naming unless necessary)